broker-core/src/test/java/io/zeebe/broker/job/JobUpdateRetriesTest.java

```diff
@@ -111,26 +112,62 @@ public void shouldRejectUpdateRetriesIfJobCompleted() {
     assertThat(response.getRecordType()).isEqualTo(RecordType.COMMAND_REJECTION);
     assertThat(response.getIntent()).isEqualTo(JobIntent.UPDATE_RETRIES);
     assertThat(response.getRejectionType()).isEqualTo(RejectionType.NOT_APPLICABLE);
-    assertThat(response.getRejectionReason()).isEqualTo("Job is not failed");
+    assertThat(response.getRejectionReason())
+        .isEqualTo(
+            "Expected to find job with key "
+                + jobEvent.getKey()
+                + ", but no job with given key exist.");
   }
 
   @Test
-  public void shouldRejectUpdateRetriesIfJobActivated() {
+  public void shouldUpdateRetriesIfJobActivated() {
     // given
     client.createJob(JOB_TYPE);
 
     apiRule.activateJobs(JOB_TYPE).await();
 
-    final Record jobEvent = client.receiveFirstJobEvent(JobIntent.ACTIVATED);
+    final Record<JobRecordValue> jobEvent = client.receiveFirstJobEvent(JobIntent.ACTIVATED);
 
     // when
     final ExecuteCommandResponse response = client.updateJobRetries(jobEvent.getKey(), NEW_RETRIES);
 
     // then
-    assertThat(response.getRecordType()).isEqualTo(RecordType.COMMAND_REJECTION);
-    assertThat(response.getIntent()).isEqualTo(JobIntent.UPDATE_RETRIES);
-    assertThat(response.getRejectionType()).isEqualTo(RejectionType.NOT_APPLICABLE);
-    assertThat(response.getRejectionReason()).isEqualTo("Job is not failed");
+    assertThat(response.getRecordType()).isEqualTo(RecordType.EVENT);
+    assertThat(response.getIntent()).isEqualTo(JobIntent.RETRIES_UPDATED);
+
+    assertThat(response.getKey()).isEqualTo(jobEvent.getKey());
+    final JobRecordValue jobEventValue = jobEvent.getValue();
+    assertThat(response.getValue())
+        .contains(
+            entry("worker", jobEventValue.getWorker()),
+            entry("type", jobEventValue.getType()),
+            entry("retries", 20L),
+            entry("deadline", jobEventValue.getDeadline().toEpochMilli()));
+
+    final Record<JobRecordValue> loggedEvent =
+        client.receiveFirstJobEvent(JobIntent.RETRIES_UPDATED);
+
+    assertThat(loggedEvent.getValue().getType()).isEqualTo(JOB_TYPE);
```

Comment:
We can remove some of the assertions, since they are tested already else where. I think would be more useful to see if the retries are really updated like `assertThat(loggedEvent.getValue().getRetries()).isEqualTo(NEW_RETRIES);`